### DISCUSSION TITLE
LED matrix module

### DISCUSSION THREAD

**Discussion by YashRajeev** [FrameworkComputer/inputmodule-rs#98] [General]
Hello I am extremely new to coding. I saw the codes that you have listed for the module. How do I input the codes?

**Comment by zcorniere** [comment]
Hello o/

You can find the protocol documentation (the file is missing some codes I think, but it is a good start) [here](https://github.com/FrameworkComputer/inputmodule-rs/blob/main/commands.md).


The protocol is really simple. You need to open the COM port of your module. On Linux it is "/dev/ttyACM0" or "/dev/ttyACM1", I don't know for Windows.

After opening the COM port you just need to write to it as if you would write to a file using the protocol documentation linked above.

For example to display a zigzag pattern on the display you need to send 4 bytes like so:
| Magic Number | Magic Number | Command | Pattern Type |
|--------|--------|--------|--------|
| 0x32 | 0x42 | 0x01 | 0x04 |

Your module will wake up and display a zig zag, until you send another command. It will go to sleep after about 60 seconds.

I'm working on [implementing this protocol in CPP](https://github.com/zcorniere/framework-imputmodule-c) (and later C) if you are more familiar with this language than Rust.
